### ISSUE TITLE
Problem encountered when spliting datas and train them separately

### ISSUE THREAD

**Issue by j40903272** [keras-team/keras#7693]
Hi all. I'm trying to build a model to generate chiense text. The problem is that when i use several atrticles as training datas and trained them each with model.fit(), i got bad results which loss does not decrease at all. However, when i combine several articles into one, it performs very well. The loss shrinks down rapidly. Besides, larger batch size seems to help either. So what causes this problem? It would need too much memory if i combine all the datas into one.....
Here's my code:

for filename in os.listdir(dir_path):
    filename = dir_path + filename
    tmp = " ".join(text_to_word_sequence(open(filename).read()))
    raw_text.append(tmp)
    # raw_text is a list with articles

seq_length = 25
model = Sequential()
model.add(LSTM(n_vocab, input_shape=(seq_length, 1), return_sequences=True))
model.add(Dropout(0.2))
model.add(LSTM(n_vocab))
model.add(Dropout(0.2))
model.add(Dense(n_vocab, activation='softmax'))
model.compile(loss='categorical_crossentropy', optimizer='adam')

dataXs, dataYs = process(raw_text)
    

for i in range(1000):
    for j in range(len(dataXs)):
        # fit the model
        model.fit(dataXs[j], dataYs[j], epochs=1, batch_size=128)

Here's what i modified which makes the performance much better:

all_text = ""
for filename in os.listdir(dir_path):
    filename = dir_path + filename
    tmp = " ".join(text_to_word_sequence(open(filename).read()))
    all_text += tmp
    
raw_text.append(all_text)

Thanks.
DADA
